reference/UiTPAS.v2.json

```diff
@@ -333,153 +353,1183 @@
+          },
+          "name": "organizerId",
+          "in": "path",
+          "required": true,
+          "description": "The id of the organizer"
+        }
+      ],
+      "get": {
+        "summary": "Get suggested financial report periods",
+        "tags": [
+          "FinancialReports"
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/ReportPeriod"
+                  }
+                },
+                "examples": {
+                  "example-1": {
+                    "value": [
+                      {
+                        "startDate": "2019-08-24",
+                        "endDate": "2019-08-24"
+                      }
+                    ]
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/Unauthorized"
+          },
+          "403": {
+            "$ref": "#/components/responses/Forbidden"
+          }
+        },
+        "operationId": "get-financials-organizer-periods",
+        "security": [
+          {
+            "USER_ACCESS_TOKEN": []
+          }
+        ],
+        "description": "Retrieve suggested report periods for an organizer."
+      }
+    },
+    "/organizers/{organizerId}/financial-reports": {
+      "parameters": [
+        {
+          "schema": {
+            "type": "string"
+          },
+          "name": "organizerId",
+          "in": "path",
+          "required": true
+        }
+      ],
+      "post": {
+        "summary": "Start an export of a financial report",
+        "operationId": "post-financials-organizers-reports",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ReportStatus"
+                },
+                "examples": {
+                  "Example of started export": {
+                    "value": {
+                      "status": "STARTED",
+                      "id": 232432
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/Unauthorized"
+          },
+          "403": {
+            "$ref": "#/components/responses/Forbidden"
+          }
+        },
+        "tags": [
+          "FinancialReports"
+        ],
+        "description": "Starts a financial report export. The result of this request is a `reportId` that can be used to request the status of the report export and the download.",
+        "security": [
+          {
+            "USER_ACCESS_TOKEN": []
+          }
+        ],
+        "parameters": [
```

Comment:
@bertramakers nu heb ik daarstraks een beetje voorvarend 2 query params toegevoegd. maar aangezien dit een POST is, is dat niet zo logisch. Ik zou ze kunnen omzetten naar form params, maar mogelijk nog beter is een request json body met ReportPeriod object?

Reply:
Yep een JSON body klinkt goed! Ik zou form params vermijden omdat het dan weer 2 formaten door elkaar worden om data door te sturen